lib/molinillo/resolution.rb

```diff
@@ -230,8 +239,13 @@ def state_index_for_unwind
         index = -1
         maximal_index = states.size - 2
         conflicts.each do |dependency_name, conflict|
-          next unless activated.vertex_named(dependency_name)
-          conflict.requirements.values.flatten(1).uniq.each do |r|
+          requirements_to_attempt_to_relax = [
+            conflict.requirement,
+            initial_requirement_for_name(dependency_name),
+            *binding_requirements_for_conflict(conflict),
+          ].compact.uniq
+
+          requirements_to_attempt_to_relax.each do |r|
```

Comment:
Sweeeeet! (as I was typing out my comment I came to realize it was probably something like that)